Parser/YSI6SeriesParse.m

```diff
@@ -1,323 +1,329 @@
-  sample_data.meta.instrument_make              = 'YSI';
-  sample_data.meta.instrument_model             = '6 Series';
-  sample_data.meta.instrument_serial_no         = '';
-  sample_data.meta.instrument_sample_interval   = median(diff(records.time*24*3600));
-  sample_data.meta.featureType                  = mode;
-  
-  sample_data.dimensions = {};
-  sample_data.variables  = {};
-
-  sample_data.dimensions{1}.name          = 'TIME';
-  sample_data.dimensions{1}.typeCastFunc  = str2func(netcdf3ToMatlabType(imosParameters(sample_data.dimensions{1}.name, 'type')));
-  sample_data.dimensions{1}.data          = sample_data.dimensions{1}.typeCastFunc(records.time');
-  
-  sample_data.variables{end+1}.name           = 'TIMESERIES';
-  sample_data.variables{end}.typeCastFunc     = str2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{end}.name, 'type')));
-  sample_data.variables{end}.data             = sample_data.variables{end}.typeCastFunc(1);
-  sample_data.variables{end}.dimensions       = [];
-  sample_data.variables{end+1}.name           = 'LATITUDE';
-  sample_data.variables{end}.typeCastFunc     = str2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{end}.name, 'type')));
-  sample_data.variables{end}.data             = sample_data.variables{end}.typeCastFunc(NaN);
-  sample_data.variables{end}.dimensions       = [];
-  sample_data.variables{end+1}.name           = 'LONGITUDE';
-  sample_data.variables{end}.typeCastFunc     = str2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{end}.name, 'type')));
-  sample_data.variables{end}.data             = sample_data.variables{end}.typeCastFunc(NaN);
-  sample_data.variables{end}.dimensions       = [];
-  sample_data.variables{end+1}.name           = 'NOMINAL_DEPTH';
-  sample_data.variables{end}.typeCastFunc     = str2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{end}.name, 'type')));
-  sample_data.variables{end}.data             = sample_data.variables{end}.typeCastFunc(NaN);
-  sample_data.variables{end}.dimensions       = [];
-
-  % convert time from seconds since 1 march 1984 00:00:00 
-  % to days since 1 jan 0000 00:00:00
-  sample_data.dimensions{1}.data = ...
-    sample_data.dimensions{1}.data / 86400 + ...
-    datenum('1-Mar-1984');
-  
-  fields = fieldnames(rmfield(records, 'time'));
-  coordinates = 'TIME LATITUDE LONGITUDE NOMINAL_DEPTH';
-  
-  % copy all the data types across to the sample_data struct
-  sample_data.variables = cell(length(fields), 1);
-  for k = 1:length(fields)
-    
-    field = records.(fields{k});
-
-    % dimensions definition must stay in this order : T, Z, Y, X, others;
-    % to be CF compliant
-    if any(strcmpi(field, {'LATITUDE', 'LONGITUDE'}))
-        sample_data.variables{end+1}.dimensions = [];
-    else
-        sample_data.variables{end+1}.dimensions = 1;
+function sample_data = YSI6SeriesParse(filename, mode)
+    %YSI6SERIESPARSE Parser for YSI 6 series MultiParameter data logger files.
```

Comment:
Again impossible to review if you change the whole indent in file at the same time you're making other changes.

Reply:
If you want to change the indent because you think it's not correct, that's fine but that's a separate task where you could do this across the whole repo in one commit.

Reply:
The problem here was that the file was not indented and the default diff algorithm/options created a confused diff. If using "-w" you will see that the only differences are newlines between flow statements (for,if, etc).